api/pkg/apis/v1alpha1/vendors/stage-vendor.go

```diff
@@ -131,6 +131,9 @@ func (s *StageVendor) Init(config vendors.VendorConfig, factories []managers.IMa
 			if err != nil {
 				sLog.Errorf("V (Stage): failed to report error status: %v (%v)", status.ErrorMessage, err)
 			}
+			// If report status succeeded, return an empty err so the subscribe function will not be retried
```

Comment:
Is this the fix of issue #300 ? Will users know the status that their activation would fail due to campaign not found? 
To me, it is OK we can even block users to create activation if campaign doesn't exist (so we can check existence in webhooks.) But if we don't want to block users to create activation before campaign, and relax the condition that campaign is created before activation is triggered (in some schedule time). At least, we need to provide the clear error status to users.

Reply:
It is fixing issue #300 . We check campaign status in the stage vendor now but the status is not updated correctly. I have fixed the logic and you can see the activation status will show below if campaign doesn't exist.

xingdong@Xingdong:~/symphony/test/integration/scenarios/04.workflow/manifest$ kubectl get activation
NAME         NEXT STAGE   STATUS
04workflow                Bad Request

status:
    errorMessage: 'Not Found: failed to get object (caused by: campaigns.workflow.symphony
      "04campaign-v1" not found)'
    isActive: true
    stage: ""
    status: 400
    statusMessage: Bad Request
    updateTime: "2024-06-11T02:34:57Z"

Reply:
I have added webhook for activations.